tests/nd_item/nd_item_async_work_group_copy_fp16.cpp

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+//
+//  SYCL 1.2.1 Conformance Test Suite
+//
+//  Copyright:	(c) 2017 by Codeplay Software LTD. All Rights Reserved.
+//
+*******************************************************************************/
+
+#include "nd_item_async_work_group_copy_common.h"
+
+#define TEST_NAME nd_item_async_work_group_copy_fp16
+
+namespace TEST_NAMESPACE {
+
+class TEST_NAME : public util::test_base {
+ public:
+  /** return information about this test
+  */
+  void get_info(test_base::info &out) const override {
+    set_test_info(out, TOSTRING(TEST_NAME), TEST_FILE);
+  }
+
+  /** execute the test
+  */
+  void run(util::logger &log) override {
+
+    auto queue = util::get_cts_object::queue();
+
+    if (!queue.get_device().has_extension("cl_khr_fp16")) {
+      log.note(
+        "Device does not support half precision floating point operations");
+      return;
+    }
+
+    check_type_and_vec<cl::sycl::half>(log);
+    check_type_and_vec<cl::sycl::cl_half>(log);
```

Comment:
And here you do not try to skip the test, which is probably good.

Reply:
Not sure I understand what you mean. You think these tests shouldn't be skipped even if cl_khr_fp16 is not supported? Or you think it's a good example for how to handle intN_t tests?